mailbox/store/src/main/java/org/apache/james/mailbox/store/StoreMessageManager.java

```diff
@@ -757,7 +757,7 @@ private MoveResult move(Iterator<MailboxMessage> originalRows, MailboxSession se
                     .messageId(messageIds.build())
                     .build(),
                 messageMoves.impactedMailboxIds().map(MailboxIdRegistrationKey::new).collect(Guavate.toImmutableSet())))
-            .then().block();
+            .blockLast();
```

Comment:
Can you explain me why?

Reply:
Well... I don't know...

My method:
 - Put `@RepeatedTest(100)` every where. It breaks.
 - Play around trying to fix things -> this is my finding.

I'm also quite surprised that it is not equivalent to be honnest.

Reply:
I guess the root cause of failed tests came from `Flux.merge()` vs `Flux.concant()`, `Flux.concat()` respects the order while `Flux.merge()` doesn't

Reply:
Not really... all the elements in the flux were not necessarily well completed, generating that very issue.

Reply:
No better solution, I've tried to put `*.defer` a bit everywhere, we have a serious concurrency issue which have to be fixed.